client/web/src/Layout.tsx

```diff
@@ -108,6 +110,7 @@ export interface LayoutProps
 
     // Search
     fetchHighlightedFileLineRanges: (parameters: FetchFileParameters, force?: boolean) => Observable<string[][]>
+    onCreateNotebook: (blocks: BlockInput[]) => void
```

Comment:
suggestion: maybe rename this to something more explicit? Since this callback isn't called *every* time a notebook is created, only in certain cases.
```suggestion
    onCreateNotebookFromNotepad: (blocks: BlockInput[]) => void
```

Reply:
Fine for me... in the long run I would hope we find a different solution that avoids passing props (as always), but that was the most straightforward solution now.